Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/baseprograms/BaseAutonomous.java

```diff
@@ -139,12 +267,115 @@ public boolean run(TelemetryPacket packet) {
             }
         };
     }
+
+    public void setMotorPower(double speed) {
+        drive.leftFront.setPower(speed);
+        drive.rightFront.setPower(speed);
+        drive.leftBack.setPower(speed);
+        drive.rightBack.setPower(speed);
+    }
+
+    public Action driveToDistanceAndMoveArm(double goalDistance, double armGoalPos) {
+        return new Action() {
+            final double epsilon = 1;
+            double iterations;
+            double goalIterations = 20;
+
+            @Override
+            public boolean run(TelemetryPacket packet) {
+                if (distSensor != null) {
+                    if (Math.abs(goalDistance - distSensor.getDistance(DistanceUnit.INCH)) < epsilon) {
+                        iterations++;
+
+                        setMotorPower(0);
+
+                        if (iterations < goalIterations)
+                            return true;
```

Comment:
Is it intended that this can end the action before the arm gets to `armGoalPos`? The number of iterations is a bit arbitrary - should it be a time-based timeout instead?